docs/operators/fixedWidthParserOperator.md

```diff
@@ -0,0 +1,205 @@
+
+**Note**: field names in the schema must match field names of the POJO and must be in the same order as it appears in the incoming data.
+
+FixedWidthParser is **idempotent**, **fault-tolerant** and **statically/dynamically partitionable**.
+
+## Class Diagram
+![](images/fixedWidthParser/fixedWidthParser.png)
+
+## Operator Information
+1. Operator location:**_malhar-contrib_**
+2. Available since:**_3.8.0_**
+3. Operator state:**_Evolving_**
+4. Java Package:[com.datatorrent.contrib.parser.FixedWidthParser](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthParser.java)
+
+
+## <a name="props"></a>Properties of FixedWidthParser
+User need to set the schema which describes fixed width data.
+
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *jsonSchema* | [Schema](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthSchema.java)  describing fixed width data. Based on schema definition the operator will parse the incoming record to object map and POJO. Valid records will be emitted as POJO / map while invalid ones are emitted on error port with error message. | String | YES | N/A |
+
+User can set this property in properties.xml in the following way
+
+```xml
+<property>
+    <name>dt.application.{ApplicationName}.operator.{OperatorName}.prop.jsonSchema</name>
+    <value>{
+      "padding": "_",
+      "alignment": "left",
+      "fields": [
+      {
+      "name": "adId",
+      "type": "Integer",
+      "length": "3",
+      "padding": "0"
+      },
+      {
+      "name": "campaignId",
+      "type": "Integer",
+      "length": "3",
+      "padding": " "
+
+      },
+      {
+      "name": "adName",
+      "type": "String",
+      "length": "10",
+      "alignment":"right"
+      },
+      {
+      "name": "bidPrice",
+      "type": "Double",
+      "length": "3"
+      },
+      {
+      "name": "startDate",
+      "type": "Date",
+      "format": "yyyy-MM-dd HH:mm:ss",
+      "length": "19"
+
+      },
+      {
+      "name": "endDate",
+      "type": "Date",
+      "format": "dd/MM/yyyy",
+      "length": "10"
+      },
+      {
+      "name": "securityCode",
+      "type": "Long",
+      "length": "5"
+      },
+      {
+      "name": "active",
+      "type": "Boolean",
+      "length": "4",
+      "trueValue": "true",
+      "falseValue": "false"
+      },
+      {
+      "name": "optimized",
+      "type": "Boolean",
+      "length": "1",
+      "trueValue": "y",
+      "falseValue": "n"
+      },
+      {
+      "name": "parentCampaign",
+      "type": "String",
+      "length": "10"
+      },
+      {
+      "name": "weatherTargeted",
+      "type": "Character",
+      "length": "1"
+      },
+      {
+      "name": "valid",
+      "type": "String",
+      "length": "10",
+      "padding":" "
```

Comment:
I have verified that for TAB user will have to give '\t' as the padding. Padding is a single character